### PR TITLE
Add pool validation to poke()

### PR DESCRIPTION
**Pull request by MaciejNadolski98**

### CODE REVIEW

File: contracts/truefi2/TrueCreditAgency.sol [hunk 1]
```diff
@@ -399,6 +399,7 @@ contract TrueCreditAgency is UpgradeableClaimable, ITrueCreditAgency {
      * @param pool Pool to update state for
      */
     function poke(ITrueFiPool2 pool) public {
+        require(isPoolAllowed[pool], "TrueCreditAgency: The pool is not whitelisted for poking");
```
Comment (MiksuJak) [inline contracts/truefi2/TrueCreditAgency.sol:402]:
```suggestion
        require(isPoolAllowed[pool], "TrueCreditAgency: The pool is not supported for poking");
```

Comment (skalermo) [inline contracts/truefi2/TrueCreditAgency.sol:402]: Add a test confirming it works as intended.

**Comment by MiksuJak** [comment]
Let's also add a test, that reaches this require

**Review by skalermo — CHANGES_REQUESTED** [review]

File: test/truefi2/lines-of-credit/TrueCreditAgency.test.ts [hunk 2]
```diff
@@ -561,6 +561,14 @@ describe('TrueCreditAgency', () => {
     })
   })
 
+  describe('poke', () => {
+    it('fails if pool is not supported', async () => {
+      await creditAgency.allowPool(tusdPool.address, false)
+      await expect(creditAgency.poke(tusd.address))
```
Comment (skalermo) [inline test/truefi2/lines-of-credit/TrueCreditAgency.test.ts:567]:
```suggestion       
      await expect(creditAgency.poke(tusdPool.address))
```

**Review by skalermo — APPROVED** [review]

**Pull request merged**